### PR TITLE
Reduce cyclomatic complexity metric

### PR DESCRIPTION
**Pull request by DavidPu**
This PR creates several helper functions to reduce cyclomatic complexity metric for  xlat_tables_map_region/unmap_region functions.

fixes arm-software/tf-issues#673

### CODE REVIEW

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by antonio-nino-diaz-arm** [comment]
This fails to build if dynamic tables are not used. Try building with ``make PLAT=fvp DEBUG=1`` for example.

File: lib/xlat_tables_v2/xlat_tables_core.c [hunk 1]
```diff
@@ -233,6 +233,98 @@ typedef enum {
 
 #if PLAT_XLAT_TABLES_DYNAMIC
 
+/*
```
Comment (antonio-nino-diaz-arm) [inline lib/xlat_tables_v2/xlat_tables_core.c]: You'll have to take this outside of the #if.

Comment (DavidPu) [reply lib/xlat_tables_v2/xlat_tables_core.c]: Thanks! fixed by taking those 2 helper functions outside of #if. I forgot to test it with PLAT_XLAT_TABLES_DYNAMIC=0.

**Comment by DavidPu** [comment]
Also added minor update to change some function parameters from 'uint32_t' to 'unsigned int' to align with latest upstream code.

**Comment by antonio-nino-diaz-arm** [comment]
jenkins: test this please

**Comment by loumay-arm** [comment]
this looks good to me

**Comment by antonio-nino-diaz-arm** [comment]
The CI has passed, it's just failing because of the renames of 2 platforms.

**Pull request merged**